homeassistant/components/waze_travel_time/__init__.py

```diff
@@ -98,10 +99,21 @@ async def async_get_travel_times_service(service: ServiceCall) -> ServiceRespons
         client = WazeRouteCalculator(
             region=service.data[CONF_REGION].upper(), client=httpx_client
         )
+
+        origin_coordinates = find_coordinates(hass, service.data[CONF_ORIGIN])
+        destination_coordinates = find_coordinates(hass, service.data[CONF_DESTINATION])
+
+        origin = origin_coordinates if origin_coordinates else service.data[CONF_ORIGIN]
+        destination = (
+            destination_coordinates
+            if destination_coordinates
+            else service.data[CONF_DESTINATION]
+        )
```

Comment:
Since waze doesn't seem to have tests for these, it's okay, but I think it would be nice to add tests so we know it doesn't break in the future